runtime/src/comm/ofi/comm-ofi.c

```diff
@@ -4188,8 +4173,13 @@ DEFN_CHPL_COMM_ATOMIC_XCHG(real64, FI_DOUBLE, _real64)
                ln, chpl_lookupFilename(fn));                            \
     chpl_comm_diags_verbose_amo("amo cmpxchg", node, ln, fn);           \
     chpl_comm_diags_incr(amo);                                          \
-    doAMO(node, object, expected, desired, result,                      \
+    Type old_value;                                                     \
+    Type old_expected;                                                  \
+    memcpy(&old_expected, expected, sizeof(Type));                      \
+    doAMO(node, object, &old_expected, desired, &old_value,             \
           FI_CSWAP, ofiType, sizeof(Type));                             \
+    *result = (chpl_bool32)(old_value == old_expected);                 \
+    memcpy(expected, &old_value, sizeof(Type));                         \
```

Comment:
The `memcpy()` is a no-op iff `*result==true`, so it might be worthwhile to if-stmt this one since in the general case the C compiler won't know the alignments or size for the copy and will have to emit general code.

Reply:
Actually, never mind that.  It does know the size, which is either 4 or 8, so it'll probably produce copy code that is no slower than a correctly-predicted branch.

Reply:
Yeah, this will generate the same code as `*expected = old_value`, I just wasn't positive there was no punning so I used a memcpy to be safe (All compilers we care about heavily optimize memcpy in this fashion.)

Reply:
Yeah, knew about memcpy() opts, just for some reason spaced that the size IS TOO known at compile time, duh.